### DISCUSSION TITLE
Complex configuration environment variable best practice

### DISCUSSION THREAD

**Discussion by travispeloton** [dasniko/testcontainers-keycloak#151] [General]
Hello! I'm playing around with the [IssuerInitializerProvider.java](https://github.com/dasniko/keycloak-extensions-demo/blob/main/initializer/src/main/java/dasniko/keycloak/initializer/issuer/IssuerInitializerProvider.java) example, and while the example sets a single issuer via `KC_SPI_INITIALIZER_ISSUER_BASE_URI` and applies it to the `master` realm ([unit tes](https://github.com/dasniko/keycloak-extensions-demo/blob/main/initializer/src/test/java/dasniko/keycloak/initializer/InitiailizerTest.java)t), the change I want to make is to set a different issuer for each different realm.

I can make this work, but was wondering what the best practice is for complex configuration. For example, instead of

```
KC_SPI_INITIALIZER_ISSUER_BASE_URI=https://auth.keycloak.de
```

I _could_ use JSON for a realm-to-issuer mapping:

```
KC_SPI_INITIALIZER_ISSUER_JSON={"master":"https://auth.keycloak.de"}
```

Good idea? other ways?

**Comment by travispeloton** [comment]
Ok, I see `ProviderConfigProperty.STRING_TYPE` was used in the example and there is a `ProviderConfigProperty.MAP_TYPE`.

Looking for how an environment variable could set it...

**Comment by dasniko** [comment]
This is completely not related to this repository!
